src/index.ts

```diff
@@ -7,15 +7,26 @@ export { outrospect };
 
 export async function createPostmanCollection(
   url: string,
-  headers?: Record<string, string>,
+  headers: Array<[string, string]>,
+  authHeader?: [string, string]
 ) {
   const introspection = await fetchIntrospection(url, headers);
-  const outrospection = outrospect(introspection);
+  const outrospection = await outrospect(introspection);
+
+  if (!outrospection || !outrospection.queries) {
+    throw new Error("Invalid outrospection data: missing 'queries' property");
+  }
+
   const queryCollection = outrospectionToQueries(outrospection);
+  // Remove the authHeader from individual request headers
+  const headersWithoutAuth = headers.filter(([key]) => authHeader ? key !== authHeader[0] : true);
```

Comment:
The purpose of this filtering is to avoid confusion. If an Auth Header is present, we should not include the same Auth Header at the request level. It should only be included at the Postman collection level.